### ISSUE TITLE
MuseScore removes accidentals if the mouse moves slightly when selecting notes

### ISSUE THREAD

**Issue by icebearisicebear** [musescore/MuseScore#21430]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

Seems like if you move your mouse even slightly when selecting a note / notes, MuseScore sees that as you're trying to drag the passage vertically, and gets rid of the accidentals in preparation for that. However that is triggered before the notes actually moves vertically. The mouse movement is big enough to make MuseScore think the user is trying to drag, but not big enough to actually move the notes vertically. This results in accidentals disappearing when user is trying to select with mouse.

### Steps to reproduce

Write a passage (or even a single note) with accidentals
Select it and move your mouse ever so slightly when selecting

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/120791885/cda0d31e-6d03-41d3-ad91-5cbe6c36bb59

### MuseScore Version

MuseScore version (64-bit): 4.2.1-240230937, revision: github-musescore-musescore-d757433

### Regression

I don't know

### Operating system

Ventura 13.6.4

**Comment by oktophonie** [comment]
Duplicate of https://github.com/musescore/MuseScore/issues/20743 (I think)
You may like to try a nightly build and see if this behaviour is better now, as it was adressed in https://github.com/musescore/MuseScore/pull/20781

**Comment by Fastball2880** [comment]
Do you click your mouse the drag after selection? It seems that in my case it does happen, but often is not. Maybe is not a Windows issue.


https://github.com/musescore/MuseScore/assets/147415340/02cb077b-a0b4-449f-a91a-49ea131e6b8c

https://github.com/musescore/MuseScore/assets/147415340/bb2c3c31-a4b3-4b96-9388-e47657867529

**Comment by icebearisicebear** [comment]
> Duplicate of #20743 (I think) You may like to try a nightly build and see if this behaviour is better now, as it was adressed in #20781

Yes I tried the nightly, the behavior is exactly the same as the release.

**Comment by icebearisicebear** [comment]
> Do you click your mouse the drag after selection? It seems that in my case it does happen, but often is not. Maybe is not a Windows issue.
> 
>  part1.mp4 
>  part2.mp4

I dont think it matters whether or not I release the mouse after selection matters, as long as I am clicking on a note or a selection and my mouse moves slightly, it does that. It might be a mac issue, I don't have Windows set up so I couldn't test it on Windows.

**Comment by dcorson-ticino-com** [comment]
>  I don't have Windows set up so I couldn't test it on Windows.

I see this same behaviour on Win10. Disappearing accidentals if you just click on a note on a shaky day.

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore version (64-bit): 4.2.1-240230937, revision: github-musescore-musescore-d757433

**Comment by bkunda** [comment]
@DmitryArefiev pls confirm whether this is a duplicate of https://github.com/musescore/MuseScore/issues/20743. Thanks!

**Comment by MarcSabatella** [comment]
> > Duplicate of #20743 (I think) You may like to try a nightly build and see if this behaviour is better now, as it was adressed in #20781
> 
> Yes I tried the nightly, the behavior is exactly the same as the release.

@icebearisicebear Did you test the 4.3 nightly or master?  Unfortunately it seems this fix has not made it into 4.3 for some reason, even though it was actually completed prior to 4.2.1.

**Comment by DmitryArefiev** [comment]
> @DmitryArefiev pls confirm whether this is a duplicate of #20743. Thanks!

Hard to say.. I was not able to reproduce #20743 on my environment

**Comment by DmitryArefiev** [comment]
> > > Duplicate of #20743 (I think) You may like to try a nightly build and see if this behaviour is better now, as it was adressed in #20781
> > 
> > 
> > Yes I tried the nightly, the behavior is exactly the same as the release.
> 
> @icebearisicebear Did you test the 4.3 nightly or master? Unfortunately it seems this fix has not made it into 4.3 for some reason, even though it was actually completed prior to 4.2.1.

It's was ported to 4.3 just recently (should be there now)

@icebearisicebear Can you try 4.3 nightly please? 

Also please attach what mouse sensitivity settings you have 
 
<img width="774" alt="Screenshot 2024-03-21 at 18 35 33" src="https://github.com/musescore/MuseScore/assets/90187801/66a34f3f-7e94-4bb0-817d-0a5e97025c78">

**Comment by zacjansheski** [comment]
Closing as stale

**Comment by cbjeukendrup** [comment]
@zacjansheski Actually I can easily reproduce this on master (if that was the question):
1. Select note with accidental
2. Drag it vertically (or hold shift and drag in any direction), but only by such a small distance that it does _not_ move to a different line (but more than 4(?) pixels, because that is the threshold for drag events to be detected)
3. The accidental is lost

Especially easy to reproduce when zooming in. 

https://github.com/musescore/MuseScore/assets/48658420/4ee354b5-70c7-4926-8b3d-1fe677ad15a7

I used to think this was caused by https://github.com/musescore/MuseScore/pull/21350, but given the open date of this issue and the merge date of that PR, that's not logical.

**Comment by cbjeukendrup** [comment]
So I think that two things should be done to resolve this issue:
- When dragging by such a small distance that the notes don't move to a different line, the accidentals should not be removed either, because that makes no sense.
- When shift+clicking or ctrl+clicking to extend a selection, that click should not additionally start dragging things, because that is hardly ever what you want.
